IMDS-13 - Fix user mapping

LotuxPunk opened the pull request.

Commit:
Fixed user mapping that could scratch some part of the existing document

Commit:
Improved mapping

Commit:
Added test for user mapping

Commit:
Fixed typo

lisatr2812 commented on lib/model/user.dart:
```diff
@@ -392,3 +410,7 @@ class User {
 }
+
+enum UserStatus { ACTIVE, DISABLED, REGISTERING }
+
```
Delete le UserType : Pour moi, cela ne devrait pas se retrouver à la vue des gens qui utilisent le sdk

lisatr2812 replied:
Et pour ne pas perdre les informations du user, dans le service updateUser (De UserApi),  tu peux faire un getUser et mettre à jour les infos de celui-ci avec ce que tu auras reçu en input, et puis faire l'update.

Commit:
Removed UserType

Commit:
Merge branch 'master' into fix/user_mapping

Commit:
Added permission constraints when updating an user

Commit:
Code cleanup
- Properly handle nullable on UserStatus
- Fixed imports
- Removed own impl. of firstOrNull

Commit:
Fixed equality issue between User entity

Commit:
Fixed equality issues on models

lisatr2812 commented on lib/api/impl/user_api_impl.dart:
```diff
@@ -18,11 +18,51 @@ class UserApiImpl extends UserApi {
+              final userToUpdate = await api.baseUserApi.getUser(modifiedUser.id);
+              final currentUser = await api.baseUserApi.getCurrentUser();
+
+              if (currentUser!.permissions.any((permission) => permission.grants.any((grant) => grant.type == PermissionItemDtoTypeEnum.ADMIN))) {
```
To make the method smaller, I will put this logic into another method, like : 
```
User updateUserFollowingCurrentUserRole(User currentUser, User userToUpdate) {
    if (currentUser!.permissions....) {
       return userToUpdate;
    } else if (currentUser.healthcarePartyId != null) {
       return userToUpdate.let((user) => {
              user.name = userToUpdate.name;
              ...
              user
       })
    } else {
       return userToUpdate.let((user) => {
              user.name = userToUpdate.name;
              ...
              user
       })
    } 
}
```

Commit:
Improved code for constraining user modifications
